src/client/common/installer.ts

```diff
@@ -189,46 +199,50 @@ export class Installer implements vscode.Disposable {
         if (SettingToDisableProduct.has(product)) {
             options.push(...[disableOption, dontShowAgain]);
         }
-        return vscode.window.showErrorMessage(`${productTypeName} ${productName} is not installed`, ...options).then(item => {
-            switch (item) {
-                case installOption: {
-                    return this.install(product);
-                }
-                case disableOption: {
-                    if (Linters.indexOf(product) >= 0) {
-                        return disableLinter(product);
-                    }
-                    else {
-                        const pythonConfig = vscode.workspace.getConfiguration('python');
-                        const settingToDisable = SettingToDisableProduct.get(product)!;
-                        return pythonConfig.update(settingToDisable, false);
-                    }
-                }
-                case useOtherFormatter: {
-                    const pythonConfig = vscode.workspace.getConfiguration('python');
-                    return pythonConfig.update('formatting.provider', alternateFormatter);
-                }
-                case dontShowAgain: {
-                    const pythonConfig = vscode.workspace.getConfiguration('python');
-                    const features = pythonConfig.get('disablePromptForFeatures', [] as string[]);
-                    features.push(productName);
-                    return pythonConfig.update('disablePromptForFeatures', features, true);
+        const item = await window.showErrorMessage(`${productTypeName} ${productName} is not installed`, ...options);
+        switch (item) {
+            case installOption: {
+                return this.install(product, resource);
+            }
+            case disableOption: {
+                if (Linters.indexOf(product) >= 0) {
+                    return this.disableLinter(product, resource).then(() => InstallerResponse.Disabled);
                 }
-                case 'Help': {
-                    return Promise.resolve();
+                else {
+                    const settingToDisable = SettingToDisableProduct.get(product)!;
+                    return this.updateSetting(settingToDisable, false, resource).then(() => InstallerResponse.Disabled);
                 }
             }
-        });
+            case useOtherFormatter: {
+                return this.updateSetting('formatting.provider', alternateFormatter, resource)
+                    .then(() => InstallerResponse.Installed);
+            }
+            case dontShowAgain: {
+                const pythonConfig = workspace.getConfiguration('python');
+                const features = pythonConfig.get('disablePromptForFeatures', [] as string[]);
+                features.push(productName);
+                return pythonConfig.update('disablePromptForFeatures', features, true).then(() => InstallerResponse.Ignore);
+            }
```

Comment:
Does TypeScript make sure that all possible case statements are covered?

Reply:
No it doesn't, will add one

Reply:
Fixed